src/common/crypto_utils/avalon_crypto_utils/worker_signing.py

```diff
@@ -0,0 +1,294 @@
+    def verify_update_receipt_signature(self, input_json):
+        """
+        Function to verify the signature of work order receipt update
+        Parameters:
+            - input_json is dictionary contains payload returned by the
+              WorkOrderReceiptUpdateRetrieve API as define EEA spec 7.2.7
+        Returns enum type SignatureStatus
+        """
+        input_json_params = input_json
+
+        concat_string = input_json_params["workOrderId"] + \
+            str(input_json_params["updateType"]) + \
+            input_json_params["updateData"]
+        concat_hash = bytes(concat_string, 'UTF-8')
+        final_hash = crypto_utility.compute_message_hash(concat_hash)
+        signature = input_json_params["updateSignature"]
+        verification_key = \
+            input_json_params["receiptVerificationKey"].encode("ascii")
+
+        decoded_signature = crypto_utility.base64_to_byte_array(signature)
+
+        return self.verify_signature_from_pubkey(decoded_signature,
+                                                 final_hash,
+                                                 verification_key)
+
+# -----------------------------------------------------------------------------
+    def verify_create_receipt_signature(self, input_json):
+        """
+        Function to verify the signature of work order receipt create
+        Parameters:
+            - input_json is dictionary contains request payload of
+              WorkOrderReceiptRetrieve API as define EEA spec 7.2.2
+        Returns enum type SignatureStatus
+        """
+        input_json_params = input_json['params']
+
+        concat_string = input_json_params["workOrderId"] + \
+            input_json_params["workerServiceId"] + \
+            input_json_params["workerId"] + \
+            input_json_params["requesterId"] + \
+            str(input_json_params["receiptCreateStatus"]) + \
+            input_json_params["workOrderRequestHash"] + \
+            input_json_params["requesterGeneratedNonce"]
+        concat_hash = bytes(concat_string, "UTF-8")
+        final_hash = bytes(crypto_utility.compute_message_hash(concat_hash))
+        signature = input_json_params["requesterSignature"]
+        verification_key = \
+            input_json_params["receiptVerificationKey"].encode("ascii")
+
+        decoded_signature = crypto_utility.base64_to_byte_array(signature)
+        return self.verify_signature_from_pubkey(decoded_signature,
+                                                 final_hash,
+                                                 verification_key)
+
+# -----------------------------------------------------------------------------
+    def verify_encryption_key_signature(
+            self, encryption_key_signature, encryption_key, verifying_key):
+        """
+        Utils function to verify integrity of worker encryption key using
+        worker verification key
+        @params encryption_key_signature - Signature computed on hash
```

Comment:
Since this a new file being introduced, ensure the function description style is consistent.

Reply:
These files are from the main Avalon repo and have not been modified. Will match the function description style in a different PR.